Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -424,31 +424,50 @@ public final class OAuth2Client {
         
         // Exchange the token
         request.send(to: self) { result in
-            guard case let .success(response) = result else {
-                completion(result)
-                return
-            }
-            
-            // Perform idToken/accessToken validation
-            do {
-                try response.result.validate(using: self, with: request as? IDTokenValidatorContext)
-            } catch {
-                completion(.failure(.validation(error: error)))
-                return
-            }
-            
             // Wait for the JWKS keys, if necessary
-            group.notify(queue: DispatchQueue.global()) {
+            group.notify(queue: DispatchQueue.global()) { [weak self] in
```

Comment:
Does this need to be a weak reference? It doesn't look like it will create a retain cycle.